src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java

```diff
@@ -722,6 +721,61 @@ public void testSimpleStream(TestContext context) {
     });
   }
 
+  private void setupTenThousandRowsTable(SQLConnection conn, Handler<AsyncResult<Void>> handler) {
+    conn.execute("BEGIN",
+      ar -> conn.execute("DROP TABLE IF EXISTS test_table",
+        ar2 -> conn.execute(CREATE_TABLE_STATEMENT,
+          ar3 -> conn.update("INSERT INTO test_table (id, name) VALUES " + TenThousandRowsData.get(),
+            ar4 -> conn.execute("COMMIT", handler)))));
+  }
+
+  @Test
+  public void testLongStream(TestContext context) {
+    Async async = context.async();
+    client.getConnection(ar -> {
+      if (ar.failed()) {
+        context.fail(ar.cause());
+        return;
+      }
+
+      // Create table
+      conn = ar.result();
+      setupTenThousandRowsTable(conn, ar2 -> {
+
+        conn.queryStream("SELECT name, id FROM test_table ORDER BY id ASC", ar3 -> {
+          if (ar3.failed()) {
+            context.fail(ar3.cause());
+          } else {
+            final SQLRowStream res = ar3.result();
+            context.assertNotNull(res);
+
+            // assert that we have columns and they are valid
+            assertNotNull(res.columns());
```

Comment:
There should be another test for streams that tests the `columns()` property. If there isn't, would you mind splitting this test? Otherwise this test tests multiple things...